packages/toolpad-app/src/appDom.ts

```diff
@@ -701,6 +700,58 @@ export function getNodeIdByName(dom: AppDom, name: string): NodeId | null {
   return index.get(name) ?? null;
 }
 
+export function getNewFirstParentIndexInParentNode(
+  dom: AppDom,
+  parent: AppDomNode,
+  parentProp: string,
+) {
+  const siblings: readonly AppDomNode[] = (getChildNodes(dom, parent) as any)[parentProp] ?? [];
+  const firstSibling = siblings.length > 0 ? siblings[0] : null;
+
+  return createFractionalIndex(null, firstSibling?.parentIndex || null);
+}
+
+export function getNewLastParentIndexInParentNode(
+  dom: AppDom,
+  parent: AppDomNode,
+  parentProp: string,
+) {
+  const siblings: readonly AppDomNode[] = (getChildNodes(dom, parent) as any)[parentProp] ?? [];
+  const lastSibling = siblings.length > 0 ? siblings[siblings.length - 1] : null;
+
+  return createFractionalIndex(lastSibling?.parentIndex || null, null);
+}
+
+export function getNewParentIndexBeforeNode(dom: AppDom, node: AppDomNode, parentProp: string) {
+  const parent = getParent(dom, node);
+
+  if (!parent) {
+    throw new Error(`Invariant: Node: "${node.id}" has no parent`);
+  }
+
+  const siblings: readonly AppDomNode[] = (getChildNodes(dom, parent) as any)[parentProp] ?? [];
```

Comment:
Wouldn't this work as well?
```suggestion
  const siblings = getSiblings(dom, node)
```

Reply:
actually the variable name should be `children` or something because i still want the node itself to be in the array - will change it to be more clear

Reply:
Is there a way we can get rid of the `as any`? Maybe witha  runtime check?

Reply:
i must've added it temporarily but then forgot to fix the types, thanks for catching that